### PR TITLE
fix: fixed bulk get missing doc incorrect result

### PR DESCRIPTION
**Pull request by HJ29**
bulk get missing doc should return object with missing status instead of throw exception

refer to couchdb bulk get sample response.
https://docs.couchdb.org/en/stable/api/database/bulk-api.html#db-bulk-get

### CODE REVIEW

**Comment by HJ29** [comment]
line 66 before changes
`var result = DocHistory.fromJson(entry!.value);`
if entry is null, non nullable exception will be thrown

File: foodb/lib/src/key_value/key_value_get.dart [hunk 1]
```diff
@@ -62,26 +62,29 @@ mixin _KeyValueGet on _AbstractKeyValue {
                 rev: d.rev?.toString() ?? 'undefined',
                 error: 'not_found',
                 reason: 'missing'));
```
Comment (yuzurihaaa) [inline foodb/lib/src/key_value/key_value_get.dart:64]: nit / non-blocker: kinda feel like can just add.
```
...
if (entry == null) {
  doc = BulkGetDoc(...)
  results.add(BulkGetIdDocs(id: d.id, docs: [doc]));
  continue;
}
... current code
```

**Comment by yuzurihaaa** [comment]
@HJ29 aha! Got it. It's to handle `null entry` because the code will still continue after the `if` block.

File: foodb/lib/src/key_value/key_value_get.dart (hunk 1 above)
Comment (HJ29) [reply foodb/lib/src/key_value/key_value_get.dart:64]: with continue statement, it  wont push the doc into result
```
if() {
  ...
  continue;
}
...
results.add(BulkGetIdDocs(id: d.id, docs: [doc])); // this line will be skipped
```
by following couchdb original bulk_get api,
https://docs.couchdb.org/en/stable/api/database/bulk-api.html#db-bulk-get
try to get doc id which doesnt exist in db should return the doc with missing status.
in couchdb example, it is the `'baz'` id sample result

**Review by lokingwei — APPROVED** [review]

**Pull request merged**